### ISSUE TITLE
Android - Release build crashes on startup

### ISSUE THREAD

**Issue by matyasf** [luca-piccioni/OpenGL.Net#23]
When I make a release build with shared runtime turned off in the build settings (I could not reproduce with debug builds) the app crashes. The issue is in GetProcAddress line 42:

```bool eglOnAndroid = Type.GetType("Android.OS.Build, Mono.Android.dll") != null;```

evaluates to false in this case. I have no idea why it happens on release builds only.

**Comment by luca-piccioni** [comment]
I wonder too. Could it be an optimization issue? The assembly is not loaded
at that point?

http://stackoverflow.com/questions/10519078/detecting-different-operating-systems-and-platforms

Maybe other checks are more portable.
Essentially you need to detect Android runtime.

Il Lun 31 Ott 2016 22:32 matyasf <notifications@github.com> ha scritto:

When I make a release build with shared runtime turned off in the build
settings (I could not reproduce with debug builds) the app crashes. The
issues is in GetProcAddress line 42:

bool eglOnAndroid = Type.GetType("Android.OS.Build, Mono.Android.dll") !=
null;

evaluates to false in this case. I have no idea why it happens on release
builds only, I could not reproduce with debug builds.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/luca-piccioni/OpenGL.Net/issues/23>, or mute the thread

**Comment by matyasf** [comment]
After lots of experimenting I've figured out how to do it:

```
        public static bool IsAndroid()
        {
            Assembly[] assems = AppDomain.CurrentDomain.GetAssemblies();
            foreach (var item in assems)
            {
                if (item.GetName().Name == "Mono.Android")
                {
                    return true;
                }
            }
            return false;
        }
```
When I replace this function call with your check it works, even when I dont use shared runtime. (I've got no idea why). Feel free to use this code; although this feels to me like a hack because I have no idea why it works, so when it could break in the future.    

Instead I suggest to either use compiler flags + ifdefs (you define the ANDROID flag in OpenGL.NET.Xamarin.Android and use an ifdef in Platform.cs) or do it like Xamarin.Forms, where the platform's value is set in the android project: Platform.cs: https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Platform.Android/Forms.cs#L123
